actionview/lib/action_view/helpers/date_helper.rb

```diff
@@ -90,12 +90,19 @@ module DateHelper
       #   distance_of_time_in_words(from_time, from_time + 50.minutes, scope: 'datetime.distance_in_words.short') # => "an hour"
       #   distance_of_time_in_words(from_time, from_time + 3.hours, scope: 'datetime.distance_in_words.short')    # => "3 hours"
       def distance_of_time_in_words(from_time, to_time = 0, options = {})
+        unless from_time.respond_to?(:to_time)
+          raise ArgumentError, "from_time must be convertible to_time"
+        end
+
+        unless to_time.respond_to?(:to_time)
+          raise ArgumentError, "from_time must be convertible to_time"
+        end
+
         options = {
           scope: :'datetime.distance_in_words'
         }.merge!(options)
 
-        from_time = from_time.to_time if from_time.respond_to?(:to_time)
-        to_time = to_time.to_time if to_time.respond_to?(:to_time)
+        from_time, to_time = from_time.to_time, to_time.to_time
         from_time, to_time = to_time, from_time if from_time > to_time
```

Comment:
@kaspth I don't think I've heard it called that before... but no, this form does not allocate an array.